src/integrationTest/java/com/mongodb/kafka/connect/MongoSourceConnectorTest.java

```diff
@@ -74,26 +82,31 @@ void testSourceLoadsDataFromMongoClient() {
         insertMany(rangeClosed(1, 50), coll1, coll2);
 
         assertAll(
-                () -> assertProduced(50, coll1),
-                () -> assertProduced(50, coll2),
-                () -> assertProduced(0, coll3));
+                () -> assertProduced(createInserts(1, 50), coll1),
```

Comment:
Updated assertProduced to explicitly test the operation types and _id values of the inserts. 

Note: The actual messages on the topics remained the same as before this patch but inspecting the messages was useful once the Kafka configuration was changed to reproduce the bug.